llvm/include/llvm/IR/GlobalValue.h

```diff
@@ -578,18 +583,23 @@ class GlobalValue : public Constant {
                                          GlobalValue::LinkageTypes Linkage,
                                          StringRef FileName);
 
+private:
   /// Return the modified name for this global value suitable to be
   /// used as the key for a global lookup (e.g. profile or ThinLTO).
   std::string getGlobalIdentifier() const;
 
-  /// Declare a type to represent a global unique identifier for a global value.
-  /// This is a 64 bits hash that is used by PGO and ThinLTO to have a compact
-  /// unique way to identify a symbol.
-  using GUID = uint64_t;
-
   /// Return a 64-bit global unique ID constructed from global value name
   /// (i.e. returned by getGlobalIdentifier()).
-  static GUID getGUID(StringRef GlobalName);
+  static GUID getGUID(StringRef GlobalIdentifier);
+
+public:
+  /// Return a 64-bit global unique ID constructed from the name of a global
+  /// symbol. Since this call doesn't supply the linkage or defining filename,
+  /// the GUID computation will assume that the global has external linkage.
+  static GUID getGUIDAssumingExternalLinkage(StringRef GlobalName) {
+    return getGUID(
+        getGlobalIdentifier(GlobalName, GlobalValue::ExternalLinkage, ""));
```

Comment:
Why add this invocation of getGlobalIdentifier which doesn't do anything other than pass through GlobalName?

Reply:
(fixed, see above)